examples/reduce.c

```diff
@@ -0,0 +1,364 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "portals4.h"
+#include "portals4_ext.h"
+
+/*
+ * This example show how to make a sum calculus with more than two knots (here four knots).
+ *
+ * ┌──────────────────┐   ┌──────────────────┐     ┌──────────────────┐   ┌──────────────────┐
+ * │                  │   │                  │     │                  │   │                  │
+ * │   md = {1,2,3}   │   │   md = {4,5,6}   │     │   md = {7,8,9}   │   │ md = {10,11,12}  │
+ * │                  │   │                  │     │                  │   │                  │
+ * └──────────────────┴─┬─┴──────────────────┘     └──────────────────┴─┬─┴──────────────────┘
+ *                      │                                               │
+ *                      │ PtlAtomic(_, PTL_SUM, _)                      │ PtlAtomic(_, PTL_SUM, _)
+ *                      │                                               │
+ *                ┌─────▼──────┐                                  ┌─────▼──────┐
+ *                │    me0 =   │                                  │    me0 =   │
+ *                │   {5,7,9}  │                                  │ {17,19,21} │
+ *                └─────┬──────┘                                  └─────┬──────┘
+ *                      │ copy to a md_handle                           │ copy to a md_handle
+ *            ┌─────────▼──────────┐                          ┌─────────▼──────────┐
+ *            │                    │                          │                    │
+ *            │    md = {5,7,9}    │                          │  md = {17,19,21}   │
+ *            │                    │                          │                    │
+ *            └────────────────────┴─────────────┬────────────┴────────────────────┘
+ *                                               │
+ *                                               │ PtlTriggeredAtomic(_, PTL_SUM, _)
+ *                                               │
+ *                                    ┌──────────▼──────────┐
+ *                                    │                     │
+ *                                    │   me1 = {22,26,30}  │
+ *                                    │                     │
+ *                                    └─────────────────────┘
+ *
+ */
+
+struct rank_info {
+	ptl_index_t pti;
+	ptl_handle_ct_t counter;
+	int size;
+	int* rank;
+	int* data;

```

Comment:
As this is an internal buffer, we can declare it as `int data[....];`

Reply:
I can do it for the rank but not for the data which is used twice in different call of rank_main.